KerbalStuff/objects.py

```diff
@@ -247,15 +248,18 @@ def __repr__(self) -> str:
 class DownloadEvent(Base):  # type: ignore
     __tablename__ = 'downloadevent'
     id = Column(Integer, primary_key=True)
-    mod_id = Column(Integer, ForeignKey('mod.id'))
+    mod_id = Column(Integer, ForeignKey('mod.id', ondelete='CASCADE'))
     mod = relationship('Mod',
                        backref=backref('downloads', order_by="desc(DownloadEvent.created)"))
```

Comment:
Whoops sorry, forgot about an SqlAlchemy quirk that prevents cascade deletes from working as is:
SqlAlchemy sets the child's foreign key columns (`mod_id` and `version_id`) to NULL _before_ it sends the DELETE for the parent, which prevents Postgres' ON DELETE CASCADE from triggering (there's no referencing row left at the time of the deletion).
We need to add `passive_deletes=True` to the backrefs, to tell SqlAlchemy to not change the foreign keys on its own, but let the database handle it.
This setting is in SqlAlchemy only, so the migration doesn't need to be updated.

Should've checked what I did in #316...

- https://docs.sqlalchemy.org/en/14/orm/relationship_api.html?highlight=passive_deletes#sqlalchemy.orm.relationship.params.passive_deletes
- https://docs.sqlalchemy.org/en/14/orm/cascades.html#passive-deletes

```suggestion
    mod_id = Column(Integer, ForeignKey('mod.id', ondelete='CASCADE'))
    mod = relationship('Mod',
                       backref=backref('downloads', passive_deletes=True, order_by="desc(DownloadEvent.created)"))
```
(same change for `version_id` again below)

Reply:
> (same change for `version_id` again below)

OK, one moment...

Reply:
OK, I think both are fixed now.